contracts/erc20-gateway/ERC20Cogateway.sol

```diff
@@ -82,4 +83,35 @@ contract ERC20Cogateway is MasterCopyNonUpgradable, GenesisERC20Cogateway, Messa
 
         activated = true;
     }
+
+    /**
+     * @notice It verifies that ERC20Gateway contract exists on origin chain
+     *         using merkle account proof.
+     *
+     * @param _blockNumber Block number at which ERC20Gateway contract is to
+     *                     be proven.
+     * @param _rlpAccount RLP encoded account node object.
+     * @param _rlpParentNodes RLP encoded value of account proof node array.
+     *
+     * \post Emits `GatewayProven` event with `messageInbox` and
+     *       `_blockNumber` parameters.
+     * \post Calls `MessageInbox.proveStorageAccount()` function with
+     *       `_blockNumber`, `_rlpAccount`, `_rlpParentNodes` as input
+     *       parameters.
+     */
+    function proveGateway(
+        uint256 _blockNumber,
+        bytes calldata _rlpAccount,
+        bytes calldata _rlpParentNodes
+    )
+        external
```

Comment:
We have `isActive` modifier in this file to check if contract is setup. Each function should have `isActive` modifier ?

Taking reference from Gnosis safe, they don't make this explicit check.  Do we need this isActive modifier? 

https://github.com/gnosis/safe-contracts/blob/aa0f3345b609a816ace6c448960ddb852b8a1bbd/contracts/GnosisSafe.sol#L126

CC @deepesh-kn

Reply:
Let's remove this for now. 
We need to implement this for almost all the auxiliary contracts.
We will think, discuss and implement it for all.